pkg/cmd/extension/manager.go

```diff
@@ -231,15 +235,8 @@ func (m *Manager) parseGitExtensionDir(fi fs.DirEntry) (Extension, error) {
 
 // getCurrentVersion determines the current version for non-local git extensions.
 func (m *Manager) getCurrentVersion(extension string) string {
-	gitExe, err := m.lookPath("git")
-	if err != nil {
-		return ""
-	}
-	dir := m.installDir()
-	gitDir := "--git-dir=" + filepath.Join(dir, extension, ".git")
-	cmd := m.newCommand(gitExe, gitDir, "rev-parse", "HEAD")
-
-	localSha, err := cmd.Output()
+	dir := filepath.Join(m.installDir(), extension)
+	localSha, err := m.gitClient.CommandOutput([]string{"rev-parse", "HEAD"}, git.WithRepoDir(dir))
```

Comment:
Wild idea: instead of passing repo dir to each individual invocation, and thus having to modify underlying GitClient methods to accept `...mods` as argument, how about adding the ability to create an instance of GitClient for a specific repo? Something like:


```go
scopedClient := gitClient.ForRepo(dir)
scopedClient.Config("remote.origin.url")
scopedClient.CheckoutBranch("branch")
```

Reply:
@mislav While I think that would work, it seems more difficult to test. 

```go
func (m *Manager) getCurrentVersion(extension string) string {
	dir := filepath.Join(m.installDir(), extension)
  scopedClient := m.gitClient.ForRepo(dir)
	localSha, err := scopedClient.CommandOutput([]string{"rev-parse", "HEAD"})
	if err != nil {
		return ""
	}
	return string(bytes.TrimSpace(localSha))
}
 
func TestGetCurrentVersion(t *testing.T) {
	tempDir := t.TempDir()
	dir := filepath.Join(tempDir, "extensions", "gh-hello")
	gc1 := &mockGitClient{}
	gc2 := &mockGitClient{}
        // We would require two mock git clients...
	gc1.On("ForRepo", dir).Return(gc2).Once()
	gc2.On("CommandOutput", []string{"rev-parse", "HEAD"}).Return("", nil).Once()
	m := newTestManager(tempDir, nil, gc1, nil)
	version := m.getCurrentVersion("gh-hello")
	assert.Equal(t, 1, version)
}
```

Reply:
I went ahead and implemented this, let me know what you think of the changes.